### PR TITLE
Feat: check cluster connectivity in distribution phase

### LINKED ISSUE #168: [Furyctl next] In distribution phase, error if kubectl version does not answer a server version

### PR DESCRIPTION
**Pull request by Al-Pragliola**
Changelist:

- added connectivity check into cluster create phase distribution by running kubectl version without flag --client 
- added connectivity check into cluster delete phase distribution by running kubectl version without flag --client 

closes #168

### CODE REVIEW

File: internal/tool/kubectl/runner.go [hunk 1]
```diff
@@ -152,8 +152,18 @@ func (r *Runner) Delete(manifestPath string, params ...string) error {
...
 
 func (r *Runner) Version() (string, error) {
+	args := []string{"version"}
+
+	if r.paths.Kubeconfig != "" {
+		args = append(args, "--kubeconfig", r.paths.Kubeconfig)
+	}
+
+	if !r.serverSide {
+		args = append(args, "--client")
```
Comment (ralgozino) [inline internal/tool/kubectl/runner.go:162]: Correct me if I'm wrong, but this logic is not correct. `serverSide` is used for example, when you do `kubectl apply --server-side`, but it is not correct to assume that if I want to do a server side apply I will also want to run `kubectl version` without the `--client` flag.

The two flags are not related.

I'd suggest separating the properties.

**Review by ralgozino — CHANGES_REQUESTED** [review]

File: internal/tool/kubectl/runner.go (hunk 1 above)
Comment (Al-Pragliola) [reply internal/tool/kubectl/runner.go:162]: _haste makes waste_

**Review by ralgozino — CHANGES_REQUESTED** [review]
From my tests, the connectivity check is being run with the `--client` flag set. So, it doesn't fail when it can't connect to the server.

See internal/tool/runner.go#78

The `--client` is useful when checking kubectl's version but we should disable it when doing the connectivity check.

Also, I'd suggest using the `--output=json` flag in the `version` command, so we get a nice parsable output :)

The --ouput=json option has been present at least since 1.20, so it is safe to use it.

**Comment by Al-Pragliola** [comment]
> From my tests, the connectivity check is being run with the `--client` flag set. So, it doesn't fail when it can't connect to the server.
> 
> See internal/tool/runner.go#78
> 
> The `--client` is useful when checking kubectl's version but we should disable it when doing the connectivity check.
> 
> Also, I'd suggest using the `--output=json` flag in the `version` command, so we get a nice parsable output :)
> 
> The --ouput=json option has been present at least since 1.20, so it is safe to use it.

added new issue about changing output in kubectl version: https://github.com/sighupio/furyctl/issues/178

**Review by ralgozino — APPROVED** [review]
LGTM

**Pull request merged**